### ISSUE TITLE
Provide @COMPILER_NAME@ in DEPENDENCY line

### ISSUE THREAD

**Issue by tkittel** [mccode-dev/McCode#1681]
Just to summarise the discussions between me and @willend today. To support NCrystal and MCPL with MSVC we obviously can't simply use flags like `-Wl,rpath,/bla` in DEPENDENCY lines. So scripts like `mcpl-config` and `ncrystal-config` must be able to provide different flags for different compilers.

The first step in that, would be for changes in mccode to ensure that the compile name is made available inside the DEPENDENCY line, e.g. like:
```
DEPENDENCY CMD(ncrystal-config --show compilerflagswithrpath @COMPILER_NAME@)
```
Step two would then be to make changes in mcpl-config and ncrystal-config to ensure the @COMPILER_NAME@ is available. It should most likely be a name from those in https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html but with the caveat that it might be changed on the fly (e.g. with `mcrun`).

**Comment by willend** [comment]
Yup. I am thinking the logic of the full adaptation of the CFLAGS line is:

1.  mcrun inserts @COMPILER_NAME@ (default is McStas build-time compiler from mccode_config, but e.g. --openacc or --mpi may influence) - perhaps also other @VARS@ 
2. ncrystal-config does its magic
3. For GPU-mode, mcrun makes a second pass on this resulting, to strip off flags (e.g. found during conda build) that `nvc` does not understand...

**Comment by tkittel** [comment]
Related to this, we should consider removing (or at least defaulting to OFF) the ENSURE_NCRYSTAL flag in McCode's CMake. Since nowadays there are easier ways to get NCrystal, so we can decouple the configuration of the two projects.

**Comment by willend** [comment]
@tkittel I believe this was in essence fixed by the introduction of @NCRYSTALFAGS@ in https://github.com/mccode-dev/McCode/pull/1855 ?
